batch/batch/aioclient.py

```diff
@@ -0,0 +1,328 @@
+            time.sleep(0.100 * j)
+            # max 4.45s
+            if i < 64:
+                i = i + 1
+
+
+class BatchClient:
+    def __init__(self, url=None, timeout=60):
+        if not url:
+            url = 'http://batch.default'
+        self.url = url
+        self._session = aiohttp.ClientSession(
+            raise_for_status=True,
+            timeout=aiohttp.ClientTimeout(total=timeout))
+
+    async def _create_job(self,  # pylint: disable=R0912
+                          image,
+                          command,
+                          args,
+                          env,
+                          ports,
+                          resources,
+                          tolerations,
+                          volumes,
+                          security_context,
+                          service_account_name,
+                          attributes,
+                          batch_id,
+                          callback,
+                          parent_ids,
+                          scratch_folder,
+                          input_files,
+                          output_files,
+                          copy_service_account_name,
+                          always_run):
+        if env:
+            env = [{'name': k, 'value': v} for (k, v) in env.items()]
+        else:
+            env = []
+        env.extend([{
+            'name': 'POD_IP',
+            'valueFrom': {
+                'fieldRef': {'fieldPath': 'status.podIP'}
+            }
+        }, {
+            'name': 'POD_NAME',
+            'valueFrom': {
+                'fieldRef': {'fieldPath': 'metadata.name'}
+            }
+        }])
+
+        container = {
+            'image': image,
+            'name': 'main'
+        }
+        if command:
+            container['command'] = command
+        if args:
+            container['args'] = args
+        if env:
+            container['env'] = env
+        if ports:
+            container['ports'] = [{
+                'containerPort': p,
+                'protocol': 'TCP'
+            } for p in ports]
+        if resources:
+            container['resources'] = resources
+        if volumes:
+            container['volumeMounts'] = [v['volume_mount'] for v in volumes]
+        spec = {
+            'containers': [container],
+            'restartPolicy': 'Never'
+        }
+        if volumes:
+            spec['volumes'] = [v['volume'] for v in volumes]
+        if tolerations:
+            spec['tolerations'] = tolerations
+        if security_context:
+            spec['securityContext'] = security_context
+        if service_account_name:
+            spec['serviceAccountName'] = service_account_name
+
+        doc = {
+            'spec': spec,
+            'parent_ids': parent_ids,
+            'always_run': always_run
+        }
+        if attributes:
+            doc['attributes'] = attributes
+        if batch_id:
+            doc['batch_id'] = batch_id
```

Comment:
This code doesn't look like it's in master for `client.py`, but is in `api.py`? Change it in `client.py` and `api.py` to be consistent.

Reply:
Oh I see, this doesn't use `api.py` anymore.

Reply:
Correct.  I think @danking made some similar simplifications in another branch.  api.py is really unnecessary.